### PR TITLE
Support stdio

### PR DESCRIPTION
**Pull request by Daniel-Liu-c0deb0t**

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -196,45 +197,72 @@ fn main() {
     }
 }
 
+fn is_stdin(args: &Args) -> bool {
+    args.files.len() == 0 || args.files[0].to_str().unwrap() == "-"
+}
+
 fn decompress_files(args: Args) {
```
Comment (krashanoff) [inline src/main.rs]: I would advise a refactor of this, lifting `is_stdin` to a member function in `impl Args`. Alternatively, we could lift the `Args::files` member to a function returning some `dyn W` where `W: Write`. Should reduce code duplication significantly either way.

File: src/main.rs [hunk 2]
```diff
@@ -196,45 +197,72 @@ fn main() {
...
-        io::copy(&mut gz_in, &mut output);
+    if is_stdin(&args) {
+        let mut gz_in = GzDecoder::new(io::stdin());
+        io::copy(&mut gz_in, &mut io::stdout()).unwrap();
+    } else {
+        for file in args.files {
+            let file_name = file.file_name().unwrap().to_str().unwrap();
+            let mut output: Box<dyn Write> = if args.to_stdout {
+                Box::new(io::stdout())
+            } else {
```
Comment (krashanoff) [inline src/main.rs]: Similarly, we can lift this behavior to some `Args::output(&mut self) -> Box<dyn Write>` to reduce duplication.

**Review by krashanoff — COMMENTED** [review]
Daniel, this looks good to me. A few suggestions on code quality.

**Comment by Daniel-Liu-c0deb0t** [comment]
Thanks for the review, Leo!

**Pull request merged**